### ISSUE TITLE
error in interpolation.py

### ISSUE THREAD

**Issue by celiotine** [POSYDON-code/POSYDON-public#286]
This error is being produced by interpolation.py for rare binary instances:

<img width="842" alt="Screen Shot 2024-03-27 at 1 23 03 PM" src="https://github.com/POSYDON-code/POSYDON/assets/13874913/042a200e-d2df-4630-bfc9-3d0f2db66e97">

This error is happening in this chunk of code in `get_final_values()`:
```
while (kvalue_low is None or np.isnan(kvalue_low)):
           mass_low = np.max(self.grid_mass[mass_low > self.grid_mass])
           try:
               kvalue_low = self.grid_final_values[mass_low][key]
           except KeyError:
               self.load_grid(mass_low)
               kvalue_low = self.grid_final_values[mass_low][key]
```
During the iteration of the `while` loop, `mass_low` becomes <= all values in `self.grid_mass`, so the cut `self.grid_mass[mass_low > self.grid_mass]` returns an empty array, and `np.max()` fails. 

Not sure if the logic needs to be fixed or if this should be caught in some sort of try/except.

Note: This error is happening for binaries out-of-hull (see Issue #265 for more info):
<img width="787" alt="Screen Shot 2024-03-27 at 1 27 11 PM" src="https://github.com/POSYDON-code/POSYDON/assets/13874913/f924ef58-378e-4eaa-89d6-dce7118faf3d">

You can reproduce this error by evolving the following binary with the posydon-error-checking branch:

```
star_1 = SingleStar(**{'mass': 23.17680615091953, 'state': 'H-rich_Core_H_burning',\
                'natal_kick_array': [190.85500146900986, 1.6094340867322998, 2.199150720317031, 1.5976269376867043]})
star_2 = SingleStar(**{'mass': 8.542829264179423, 'state': 'H-rich_Core_H_burning',\
                'natal_kick_array': [None, None, None, None]})

binary = BinaryStar(star_1, star_2,
                   **{'time': 0.0, 
                      'state': 'detached', 
                      'event': 'ZAMS',
                      'orbital_period': 39.47002178459404, 
                      'eccentricity': 0.0},
                    properties = sim_prop)

binary.evolve()
```

**Comment by astroJeff** [comment]
@prinse1545, will you have a chance to look into this at all?

**Comment by philipp-rajah** [comment]
@astroJeff I haven't done more than look at the very descriptive report provided by Camille and it's unclear to me how to proceed. My understanding is that the actual error is being raised outside the IFInterpolation, namely in what I think is the single star interpolation part of POSYDON. Presumably this is being caused by errors introduced by the 1NN approximation. If this is the case then I assume we need more tools to understand why this is happening in order to determine whether it is acceptable. In which case if it is I can edit the code so it doesn't throw an error, but instead throws a warning or continues the evolution in the most appropriate way (if possible).

From my end I need to know what specifically is needed for understanding why this happens in an astrophysical sense and then what is the most appropriate way to handle this from your perspective i.e just accept it and throw a warning / error that is more descriptive or continue the evolution of possible. As for actually answering your question, yes I can look into this but I am spread quite thin across research, classes, and some tasks I'm handling for Mathias, but if we could find a time to meet and get all of this straightened out I don't think this should take too long to resolve.

**Comment by maxbriel** [comment]
I've come across this too recently. I looked into the specific `key` it is trying to match. Although I don't remember it exactly, I believe it was trying to match something like the `core_mass_at_He_depletion`. But those values are NaN if the star doesn't reach it. Or something similar.
So `mass_low` is set to the maximum mass from the `self.grid_mass[mass_low > self.grid_mass]`, but the kvalue is `NaN`. It keeps selecting the next one, but those are lower masses which will also have `NaN`s. Eventually, emptying the `grid_mass` array.

**Comment by celiotine** [comment]
This should have been resolved with PR #385. I re-ran the failing binaries above with the current development branch and they are now evolving successfully.
